tests/integration.rs

```diff
@@ -5931,6 +5959,96 @@ mod tests {
 
             handle_child_output(r, &output);
         }
+
+        #[test]
+        fn test_windows_guest_netdev_hotplug() {
+            let mut windows = WindowsDiskConfig::new(WINDOWS_IMAGE_NAME.to_string());
+            let guest = Guest::new(&mut windows);
+
+            let tmp_dir = TempDir::new_with_prefix("/tmp/ch").unwrap();
+            let mut workload_path = dirs::home_dir().unwrap();
+            workload_path.push("workloads");
+
+            let mut ovmf_path = workload_path.clone();
+            ovmf_path.push(OVMF_NAME);
+
+            let mut osdisk_path = workload_path;
+            osdisk_path.push(WINDOWS_IMAGE_NAME.to_string());
+
+            let api_socket = temp_api_path(&tmp_dir);
+
+            let mut child = GuestCommand::new(&guest)
+                .args(&["--api-socket", &api_socket])
+                .args(&["--cpus", "boot=2,kvm_hyperv=on"])
+                .args(&["--memory", "size=2G,hotplug_size=5G"])
+                .args(&["--kernel", ovmf_path.to_str().unwrap()])
+                .default_disks()
+                .args(&["--serial", "tty"])
+                .args(&["--console", "off"])
+                .args(&["--net", "tap="])
+                .capture_output()
+                .spawn()
+                .unwrap();
+
+            // Wait to make sure Windows boots up
+            thread::sleep(std::time::Duration::new(60, 0));
```

Comment:
This question is not related to this PR change.

Is there a (easy) way to adopt `wait_vm_boot()` on windows guests (instead of blindly waiting)? We will need to provision the windows guest with something similar to what we did on linux guests (through cloud-init [here](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/master/test_data/cloud-init/ubuntu/user-data#L66-L72)).

The context is I see the windows CI now is taking more than 12 minutes (to run 5 tests sequentially), and it could be a bottleneck for our CI pipeline when we add more windows tests.

Reply:
Good catch! Yep, polling on something with timeout will be a better solution. I'll create an issue for this, perhaps should be done in the scope of test refactoring i mentioned in the description.

Thanks!